api/notd/ownership_manager.py

```diff
@@ -55,70 +57,72 @@ async def update_token_ownership(self, registryAddress: str, tokenId: str):
+                else:
+                    await self.saver.create_token_ownership(connection=connection, registryAddress=retrievedTokenOwnership.registryAddress, tokenId=retrievedTokenOwnership.tokenId, ownerAddress=retrievedTokenOwnership.ownerAddress, transferDate=retrievedTokenOwnership.transferDate, transferValue=retrievedTokenOwnership.transferValue, transferTransactionHash=retrievedTokenOwnership.transferTransactionHash)
 
     @staticmethod
     def _uniqueness_tuple_from_token_multi_ownership(retrievedTokenMultiOwnership: RetrievedTokenMultiOwnership) -> Tuple[str, str, str, int, int, datetime.datetime, str]:
         return (retrievedTokenMultiOwnership.registryAddress, retrievedTokenMultiOwnership.tokenId, retrievedTokenMultiOwnership.ownerAddress, retrievedTokenMultiOwnership.quantity, retrievedTokenMultiOwnership.averageTransferValue, retrievedTokenMultiOwnership.latestTransferDate, retrievedTokenMultiOwnership.latestTransferTransactionHash)
 
     async def _update_token_multi_ownership(self, registryAddress: str, tokenId: str) -> None:
-        registryAddress = chain_util.normalize_address(value=registryAddress)
-        latestTransfers = await self.retriever.list_token_transfers(fieldFilters=[
-            StringFieldFilter(fieldName=TokenTransfersTable.c.registryAddress.key, eq=registryAddress),
-            StringFieldFilter(fieldName=TokenTransfersTable.c.tokenId.key, eq=tokenId),
-        ], orders=[Order(fieldName=BlocksTable.c.updatedDate.key, direction=Direction.DESCENDING)], limit=1)
-        if len(latestTransfers) == 0:
-            return
-        latestTransfer = latestTransfers[0]
-        latestOwnerships = await self.retriever.list_token_multi_ownerships(fieldFilters=[
-            StringFieldFilter(fieldName=TokenMultiOwnershipsTable.c.registryAddress.key, eq=registryAddress),
-            StringFieldFilter(fieldName=TokenMultiOwnershipsTable.c.tokenId.key, eq=tokenId),
-        ], orders=[Order(fieldName=TokenMultiOwnershipsTable.c.updatedDate.key, direction=Direction.DESCENDING)], limit=1)
-        latestOwnership = latestOwnerships[0] if len(latestOwnerships) > 0 else None
-        if latestOwnership is not None and latestOwnership.updatedDate > latestTransfer.updatedDate:
-            logging.info(f'Skipping updating token_multi_ownership because last record ({latestOwnership.updatedDate}) is newer that last transfer update ({latestTransfer.updatedDate})')
-            return
-        retrievedTokenMultiOwnerships = await self.tokenOwnershipProcessor.calculate_token_multi_ownership(registryAddress=registryAddress, tokenId=tokenId)
-        async with self.saver.create_transaction() as connection:
-            currentTokenMultiOwnerships = await self.retriever.list_token_multi_ownerships(connection=connection, fieldFilters=[
+        async with self.lockManager.with_lock(name=f"{registryAddress}-{tokenId}-lock", timeoutSeconds=5, expirySeconds=60):
```

Comment:
```suggestion
        async with self.lockManager.with_lock(name=f"update-multi-ownership-{registryAddress}-{tokenId}", timeoutSeconds=30, expirySeconds=300):
```

Reply:
also, i think this lock can be moved much lower - we should lock as little as possible. you should only be locking around the transaction below i believe